evalml/pipelines/component_graph.py

```diff
@@ -691,13 +697,18 @@ def describe(self, return_dict=False):
         for number, component in enumerate(self.component_instances.values(), 1):
             component_string = str(number) + ". " + component.name
             logger.info(component_string)
-            components.update(
-                {
-                    component.name: component.describe(
-                        print_name=False, return_dict=return_dict
-                    )
-                }
-            )
+            try:
+                components.update(
+                    {
+                        component.name: component.describe(
+                            print_name=False, return_dict=return_dict
+                        )
+                    }
+                )
+            except TypeError:
+                raise TypeError(
```

Comment:
I think a different error would be more useful here, a TypeError doesn't necessarily make sense for illegally calling a method. Personally, I always default to a ValueError.